lib/plugins/aws/deploy/compile/events/apiGateway/index.js

```diff
@@ -36,7 +36,8 @@ class AwsCompileApigEvents {
         _.forEach(this.serverless.service.functions, functionObj => {
           if (functionObj.events) {
             functionObj.events.forEach(event => {
-              if (event.http) noEndpoints = false;
+              // Allow events with empty http event to validate function
+              if ({}.hasOwnProperty.call(event, 'http')) noEndpoints = false;
```

Comment:
What's the benefit of using this `{}.hasOwnProperty.call(event, 'http')` syntax over `event.hasOwnProperty('http')`? It seems a bit opaque, so I'm hoping there's a specific reason.